stage0/gc.go

```diff
@@ -73,9 +74,15 @@ type mounts []*mount
 // parent. This works by pushing all child mounts up to the front. Parent
 // mounts are halted once they encounter a child. The actual sequential order
 // of the mounts doesn't matter; only that children are in front of parents.
-func (m mounts) Less(i, j int) bool { return m[i].id != m[j].parentId }
-func (m mounts) Len() int           { return len(m) }
-func (m mounts) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
+// func (m mounts) Less(i, j int) bool { return m[i].id != m[j].parentId && m[i].parentId > m[j].parentId }
+func (m mounts) Less(i, j int) bool {
+	isNotParent := m[i].id != m[j].parentId
+	hasHigherParent := m[i].parentId > m[j].parentId
```

Comment:
I don't think comparing ids always work in the general case.

Let's say you have this tree:
```
Id1
├─ Id4
│   └ directory
├─ Id2
│  ├ Id3
```

So far, the ids are in order.

And then, someone did a `mount("/Id1/Id4/Id4", "/Id1/Id/directory", MS_MOVE)`.

```
Id1
├─ Id4
│   └ Id3
├─ Id2
│  ├ 
```

Id3 is now a child of Id4 even though it has a lower id.